testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveSparkCompatibility.java

```diff
@@ -187,6 +187,51 @@ public void testReadSparkCreatedTable(String sparkTableFormat, String expectedTr
         onSpark().executeQuery("DROP TABLE " + sparkTableName);
     }
 
+    @Test(groups = {HIVE_SPARK, PROFILE_SPECIFIC_TESTS}, dataProvider = "sparkParquetTimestampFormats")
+    public void testSparkParquetTimestampCompatibility(String sparkTimestampFormat, String sparkTimestamp, String trinoTimestampPrecision, java.sql.Timestamp expected)
+    {
+        String sparkTableName = "test_spark_parquet_timestamp_compatibility_" + sparkTimestampFormat.toLowerCase(ENGLISH) + "_" + trinoTimestampPrecision + "_" + randomTableSuffix();
+        String trinoTableName = format("%s.default.%s", TRINO_CATALOG, sparkTableName);
+
+        onSpark().executeQuery("SET spark.sql.parquet.outputTimestampType = " + sparkTimestampFormat);
+        onSpark().executeQuery(
+                "CREATE TABLE default." + sparkTableName + "(a_timestamp timestamp) " +
+                        "USING PARQUET " +
+                        "TBLPROPERTIES ('transactional'='false')");
+
+        onSpark().executeQuery("INSERT INTO " + sparkTableName + " VALUES (TIMESTAMP '" + sparkTimestamp + "')");
+
+        onTrino().executeQuery("SET SESSION hive.timestamp_precision = '" + trinoTimestampPrecision + "'");
+        Row expectedRow = row(expected);
+        assertThat(onTrino().executeQuery("SELECT * FROM " + trinoTableName)).containsOnly(expectedRow);
+        assertThat(onTrino().executeQuery("SELECT count(*) FROM " + trinoTableName + " WHERE a_timestamp = TIMESTAMP '" + expected.toString() + "'")).containsOnly(row(1));
```

Comment:
would prefer to make `expected` a plan `String` rather than have to check whether `java.sql.Timestamp#toString` does something stupid.